test/integration/QueryRunnerTest.kt

```diff
@@ -0,0 +1,79 @@
+
+import com.vaticle.typedb.client.api.TypeDBSession
+import com.vaticle.typedb.client.api.TypeDBTransaction
+import com.vaticle.typedb.studio.framework.material.Icon
+import com.vaticle.typedb.studio.state.StudioState
+import com.vaticle.typedb.studio.state.common.util.Label
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.Delays
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.clickIcon
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.clickText
+import com.vaticle.typedb.studio.test.integration.data.Paths.SampleGitHubData
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.connectToTypeDB
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.copyFolder
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.createDatabase
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.delayAndRecompose
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.openProject
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.writeDataInteractively
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.writeSchemaInteractively
+import com.vaticle.typedb.studio.test.integration.common.TypeDBRunners.withTypeDB
+
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+
+class QueryRunnerTest: IntegrationTest() {
+
+    @Test
+    fun inferNotExplainRegression() {
+        withTypeDB { typeDB ->
+            runBlocking {
+                connectToTypeDB(composeRule, typeDB.address())
+                createDatabase(composeRule, dbName = testID)
+                copyFolder(source = SampleGitHubData.path, destination = testID)
+                openProject(composeRule, projectDirectory = testID)
+                writeSchemaInteractively(composeRule, dbName = testID, SampleGitHubData.schemaFile)
+                writeDataInteractively(composeRule, dbName = testID, SampleGitHubData.dataFile)
+
+                StudioState.client.session.tryOpen(database = testID, TypeDBSession.Type.DATA)
+                delayAndRecompose(composeRule, Delays.NETWORK_IO)
+
+                StudioState.project.current!!.directory.entries.find {
+                    it.name == SampleGitHubData.collaboratorsQueryFile
+                }!!.asFile().tryOpen()
+
+                clickText(composeRule, Label.DATA.lowercase())
+                clickText(composeRule, Label.READ.lowercase())
+                clickText(composeRule, Label.INFER.lowercase())
+
+                StudioState.pages.active?.let { if (it.isRunnable) it.asRunnable().mayOpenAndRun() }
+
+                val priorTransaction = StudioState.client.session.transaction.transaction!!
+                val priorTransactionInfer = priorTransaction.options().infer().get()
+                val priorTransactionExplain = priorTransaction.options().explain().get()
+
+                assert(priorTransactionInfer && !priorTransactionExplain)
```

Comment:
Weren't we going to make this test more useful by checking that _all_ the options match the expected values?

Otherwise, the chance of this test ever breaking is extremely low, as the code writer would have to make _exactly_ the same mistake as before.

Reply:
Good idea. I've rescoped and updated the test to reflect this.